modules/generic-scala3/src/test/scala-3/pureconfig/generic/ProductHintSuite.scala

```diff
@@ -0,0 +1,217 @@
+package pureconfig
+
+import scala.collection.JavaConverters.*
+
+import com.typesafe.config.{ConfigFactory, ConfigObject, ConfigValueType}
+
+import pureconfig.error.*
+import pureconfig.generic.ProductHint
+import pureconfig.generic.semiauto.deriveReader
+import pureconfig.syntax.*
+
+class ProductHintSuite extends BaseSuite {
+
+  behavior of "ProductHint"
+
+  case class ConfWithCamelCaseInner(thisIsAnInt: Int, thisIsAnotherInt: Int)
+  case class ConfWithCamelCase(camelCaseInt: Int, camelCaseString: String, camelCaseConf: ConfWithCamelCaseInner)
+
+  val confWithCamelCase = ConfWithCamelCase(1, "foobar", ConfWithCamelCaseInner(2, 3))
+
+  /** return all the keys in a `ConfigObject` */
+  def allKeys(configObject: ConfigObject): Set[String] = {
+    configObject.toConfig().entrySet().asScala.flatMap(_.getKey.split('.')).toSet
+  }
+
+  it should "read kebab case config keys to camel case fields by default" in {
+    given ConfigReader[ConfWithCamelCase] = deriveReader
+
+    val conf = ConfigFactory.parseString("""{
+      camel-case-int = 1
+      camel-case-string = "bar"
+      camel-case-conf {
+        this-is-an-int = 3
+        this-is-another-int = 10
+      }
+    }""")
+
+    conf.to[ConfWithCamelCase] shouldBe Right(ConfWithCamelCase(1, "bar", ConfWithCamelCaseInner(3, 10)))
+  }
```

Comment:
Is there a reason why you chose not to add name conversion tests like "write kebab case config keys..." and "write camel case config keys..." in `modules/generic/src/test/scala/pureconfig/ProductHintSuite.scala`?

Reply:
Those tests require `ConfigWriter` which derivation is not implemented in this PR. I'll add it in some subsequent one.